docroot/modules/custom/uiowa_core/src/EntityItemProcessorBase.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\uiowa_core;
+
+/**
+ * A base for node processing classes.
+ */
+abstract class EntityItemProcessorBase {
+
+  /**
+   * The map of entity field names to source field names.
+   *
+   * @var array
+   */
+  protected static $fieldMap = [];
+
+  /**
+   * Process an individual entity.
+   */
+  public static function process($entity, $record): bool {
+    $updated = FALSE;
+    foreach (static::$fieldMap as $to => $from) {
+      // @todo Add a message if a node doesn't have a field.
+      if ($entity->hasField($to) && $entity->get($to)->value != $record->{$from}) {
+        $entity->set($to, $record->{$from});
+        $updated = TRUE;
+      }
```

Comment:
List fields need a little different handling, and with this current set up will always end up with a value of `null` being compared to the record's value. Currently with how classrooms is set up, it's leading to the process thinking every node has been updated (though after the data is set and a save attempted, it's not recording a revision because there are no actual changes being made).

Reply:
We probably need to check if the `$record->{$from}` is an array or object and process it differently if so. What is the field that this pertains to?

Reply:
For classrooms, it's
`field_room_responsible_unit	`, `field_room_features`, `field_room_technology_features `, `field_room_type`, and `field_room_scheduling_regions` that it effects.

Reply:
The way I imagined this happening is that in the `processRecord` step, the value on `$record` gets replaced with a `target_id`.

Reply:
Ah, I'm getting the issue now. I think for the field, we can check the field definition that is attached to it and see if its an ER field. So more forked if/then logic...

Reply:
Cool. I'm going to look at this and maybe make a few minor adjustments.

Reply:
It seems like we still might need to account for multiple values differently, but I think the code we have now is working for both use cases as is.